cargo-dist/src/backend/installer/homebrew.rs

```diff
@@ -56,64 +40,87 @@ pub struct HomebrewInstallerInfo {
-    use_sha256_checksum(manifest, &info.x86_64_macos, &mut info.x86_64_macos_sha256);
-    use_sha256_checksum(manifest, &info.arm64_linux, &mut info.arm64_linux_sha256);
-    use_sha256_checksum(manifest, &info.x86_64_linux, &mut info.x86_64_linux_sha256);
+    let info = info.clone();
 
-    let script = dist
-        .templates
-        .render_file_to_clean_string(TEMPLATE_INSTALLER_RB, &info)?;
-    LocalAsset::write_new(&script, &info.inner.dest_path)?;
-    Ok(())
-}
-
-/// Grab the sha256 checksum for this artifact from the manifest
-fn use_sha256_checksum(
-    manifest: &DistManifest,
-    fragment: &Option<ExecutableZipFragment>,
-    checksum: &mut Option<String>,
-) {
     let checksum_key = ChecksumStyle::Sha256.ext();
-    if let Some(frag) = &fragment {
-        *checksum = manifest
+    let map_fragment = |fragment: ExecutableZipFragment| -> HomebrewFragment {
+        let sha256 = manifest
             .artifacts
-            .get(&frag.id)
+            .get(&fragment.id)
             .and_then(|a| a.checksums.get(checksum_key))
-            .cloned()
-    }
-}
+            .cloned();
+        let linkage = manifest.linkage_for_artifact(&fragment.id);
 
-// Grab the linkage for this artifact from the manifest
-fn use_linkage(
-    manifest: &DistManifest,
-    fragment: &Option<ExecutableZipFragment>,
-    dependencies: &mut Vec<String>,
-) {
-    if let Some(frag) = fragment {
-        let archive_linkage = manifest.linkage_for_artifact(&frag.id);
-        let homebrew_deps = archive_linkage
+        let dependencies = linkage
             .homebrew
             .iter()
-            .filter_map(|lib| lib.source.clone());
-        dependencies.extend(homebrew_deps);
+            .filter_map(|lib| lib.source.clone())
+            .collect();
+
+        HomebrewFragment {
+            fragment,
+            sha256,
+            dependencies,
+        }
+    };
+
+    macro_rules! map_fragments {
+        ($fragments:ident = ($($name:ident),*)) => {
+            let $fragments = HomebrewFragments {
+                $($name: $fragments.$name.clone().map(map_fragment)),*
+            };
+        };
     }
+    map_fragments!(fragments = (arm64_linux, x86_64_linux, arm64_macos, x86_64_macos));
+
+    let dest_path = info.inner.dest_path.clone();
+    let inputs = HomebrewTemplateInputs { info, fragments };
+
+    let script = dist
+        .templates
+        .render_file_to_clean_string(TEMPLATE_INSTALLER_RB, &inputs)?;
+    LocalAsset::write_new(&script, dest_path)?;
+    Ok(())
+}
+
+#[derive(Debug, Clone, Serialize)]
+struct HomebrewTemplateInputs {
+    #[serde(flatten)]
+    info: HomebrewInstallerInfo,
+
+    #[serde(flatten)]
+    fragments: HomebrewFragments<HomebrewFragment>,
+}
+
+#[derive(Debug, Clone, Serialize)]
+struct HomebrewFragment {
+    #[serde(flatten)]
+    fragment: ExecutableZipFragment,
+
+    /// SHA256 sum of the fragment — if we have it, which we might not, in tests at least?
```

Comment:
The local builds usually happen on a different host from the global build, so having these depends on the sha256s getting piped through all the way from the local build manifests to the info we have at the global step. We do that in CI, but it's not guaranteed that someone doing their own kinda builds outside this environment would have them.

(Same deal if you're just generating the installers via `dist build -aglobal` without having actually built any local artifacts at all.)